src/controllers/networks/networks.ts

```diff
@@ -0,0 +1,265 @@
+import EmittableError from '../../classes/EmittableError'
+import { networks as predefinedNetworks } from '../../consts/networks'
+import {
+  AddNetworkRequestParams,
+  Network,
+  NetworkId,
+  NetworkInfo,
+  NetworkInfoLoading
+} from '../../interfaces/network'
+import { Storage } from '../../interfaces/storage'
+import { getFeaturesByNetworkProperties, getNetworkInfo } from '../../libs/networks/networks'
+import EventEmitter, { Statuses } from '../eventEmitter/eventEmitter'
+
+const STATUS_WRAPPED_METHODS = {
+  addNetwork: 'INITIAL',
+  updateNetwork: 'INITIAL'
+} as const
+
+export class NetworksController extends EventEmitter {
```

Comment:
Please when you have some time, add a few notes for this controller (that sums up its purpose).